### PR TITLE
Update cilium to v1.13.0

### PR DESCRIPTION
**Pull request by DockToFuture**
**How to categorize this PR?**

/area networking
/kind enhancement

**What this PR does / why we need it**:
Update cilium to `v1.13.0`.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
Update cilium to `v1.13.0`.
```

### CODE REVIEW

**Comment by axel7born** [comment]
If we don't merge the new init container [`config`](https://github.com/cilium/cilium/blob/469bcb18b44002ca34254b26de4a6c2b19868e18/install/kubernetes/cilium/templates/cilium-agent/daemonset.yaml#L403) now, we should probably take an other look later. It seems to be part of the implementation of an interesting feature, that allows to have per node configuration (https://github.com/cilium/cilium/issues/22036).
The new `role` and `rolebinding` are probably only needed for that container.

File: charts/internal/cilium/charts/agent/templates/rolebinding.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  labels:
+    app.kubernetes.io/part-of: cilium
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: cilium-config-agent
+subjects:
+  - kind: ServiceAccount
+    name: "cilium"
+    namespace: {{ .Release.Namespace }}
```
Comment (axel7born) [inline charts/internal/cilium/charts/agent/templates/rolebinding.yaml:16]: Probably not needed, when we omit the new init container.

**Comment by DockToFuture** [comment]
I've added the init container since it is offering some great features as mentioned above.

**Review by axel7born — APPROVED** [review]

**Pull request merged**